packages/test-base/src/androidAndroidTest/kotlin/io/realm/kotlin/test/shared/SerializationTests.kt

```diff
@@ -0,0 +1,649 @@
+                        nullableRealmAnyField = RealmAny.create(shortField)
+                    }
+                    Int::class -> SerializableSample().apply {
+                        nullableRealmAnyField = RealmAny.create(intField)
+                    }
+                    Long::class -> SerializableSample().apply {
+                        nullableRealmAnyField = RealmAny.create(longField)
+                    }
+                    Float::class -> SerializableSample().apply {
+                        nullableRealmAnyField = RealmAny.create(floatField)
+                    }
+                    Double::class -> SerializableSample().apply {
+                        nullableRealmAnyField = RealmAny.create(doubleField)
+                    }
+                    ByteArray::class -> SerializableSample().apply {
+                        nullableRealmAnyField = RealmAny.create(binaryField)
+                    }
+                    Boolean::class -> SerializableSample().apply {
+                        nullableRealmAnyField = RealmAny.create(booleanField)
+                    }
+                    String::class -> SerializableSample().apply {
+                        nullableRealmAnyField = RealmAny.create(stringField)
+                    }
+                    Decimal128::class -> SerializableSample().apply {
+                        nullableRealmAnyField = RealmAny.create(decimal128Field)
+                    }
+                    RealmInstant::class -> SerializableSample().apply {
+                        nullableRealmAnyField = RealmAny.create(timestampField)
+                    }
+                    ObjectId::class -> SerializableSample().apply {
+                        nullableRealmAnyField = RealmAny.create(objectIdField.asBsonObjectId())
+                    }
+                    RealmUUID::class -> SerializableSample().apply {
+                        nullableRealmAnyField = RealmAny.create(uuidField)
+                    }
+                    RealmObject::class -> SerializableSample().apply {
+                        nullableObject = SerializableSample()
+                        nullableRealmAnyField = RealmAny.create(nullableObject!!)
+                    }
+                    else -> throw IllegalStateException("Untested type $classifier")
+                }
+            }
+            .forEach { expected ->
+                val encoded: String = json.encodeToString(expected)
+                val decoded: SerializableSample = json.decodeFromString(encoded)
+
+                RealmAny::class.assertValue(expected.nullableRealmAnyField, decoded.nullableRealmAnyField)
+            }
+    }
+
+    private fun List<CollectionTypeSafetyManager<Any?>>.exhaustiveCollectionTesting() =
+        forEach { dataset ->
+            listOf(
+                dataset.createPrePopulatedContainer(),
+                realm.writeBlocking {
+                    dataset.createPrePopulatedContainer()
+                }
```

Comment:
Not sure I understand this, same with dictionaries below.

Reply:
We are using a list to reuse the logic in the `forEach` block.